velox/exec/Window.cpp

```diff
@@ -41,16 +41,69 @@ void initKeyInfo(
   }
 }
 
-void checkDefaultWindowFrame(const core::WindowNode::Function& windowFunction) {
-  VELOX_CHECK_EQ(
-      windowFunction.frame.type, core::WindowNode::WindowType::kRange);
-  VELOX_CHECK_EQ(
-      windowFunction.frame.startType,
-      core::WindowNode::BoundType::kUnboundedPreceding);
-  VELOX_CHECK_EQ(
-      windowFunction.frame.endType, core::WindowNode::BoundType::kCurrentRow);
-  VELOX_CHECK_EQ(windowFunction.frame.startValue, nullptr);
-  VELOX_CHECK_EQ(windowFunction.frame.endValue, nullptr);
+vector_size_t findFrameEndPointUnboundedPreceding(
+    core::WindowNode::WindowType /*type*/,
+    vector_size_t /*functionNumber*/,
+    vector_size_t partitionStartRow,
+    vector_size_t /*partitionEndRow*/,
+    vector_size_t /*currentRow*/) {
+  return partitionStartRow;
+}
+
+vector_size_t findFrameEndPointCurrentRowRangeMode(
+    core::WindowNode::WindowType type,
+    vector_size_t /*functionNumber*/,
+    vector_size_t /*partitionStartRow*/,
+    vector_size_t /*partitionEndRow*/,
+    vector_size_t currentRow) {
+  return currentRow;
+}
+
+vector_size_t findFrameEndPointUnboundedFollowing(
+    core::WindowNode::WindowType /*type*/,
+    vector_size_t /*functionNumber*/,
+    vector_size_t /*partitionStartRow*/,
+    vector_size_t partitionEndRow,
+    vector_size_t /*currentRow*/) {
+  return partitionEndRow;
+}
+
+windowFrameFunctionPtr getWindowFrameStart(
+    core::WindowNode::BoundType startType,
+    core::WindowNode::WindowType type) {
+  switch (startType) {
+    case core::WindowNode::BoundType::kUnboundedPreceding:
+      return findFrameEndPointUnboundedPreceding;
+    case core::WindowNode::BoundType::kCurrentRow: {
+      if (type == core::WindowNode::WindowType::kRange) {
+        return findFrameEndPointCurrentRowRangeMode;
+      }
+    }
+    case core::WindowNode::BoundType::kPreceding:
+    case core::WindowNode::BoundType::kFollowing:
+    case core::WindowNode::BoundType::kUnboundedFollowing:
```

Comment:
Can UnboundedFollowing be used for the frame start ? Similarly can Unbounded Preceding be used for frame end ? Please can you check how Presto java handles these.